ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -135,6 +140,18 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
           exception = e;
         }
       }
+    } else if (restRequest.getRestMethod() == RestMethod.POST && storageQuotaService != null
+        && restRequest.getSize() > 0) {
+      try {
+        Account account = getAccountFromArgs(restRequest.getArgs());
+        Container container = getContainerFromArgs(restRequest.getArgs());
+        if (storageQuotaService.shouldThrottle(account.getId(), container.getId(), QuotaOperation.Post,
+            restRequest.getSize())) {
```

Comment:
Nit: add a comment here that we are optimistically accounting for the request size beforehand and substracting it from quota. This might have edge cases if the request actually fails later.